lib/kazan/watcher.ex

```diff
@@ -146,6 +159,21 @@ defmodule Kazan.Watcher do
     {:ok, state}
   end
 
+  @impl GenServer
+  def handle_cast(:stop_watch, %State{id: request_id} = state) do
+    Logger.info("Stopping watch #{inspect(self())}")
```

Comment:
Should we be using `state.debug` and `Logger.debug` for this like we're doing for the other log messages?

Reply:
Actually I didn't understand the purpose of having the `debug` function - why not just have `Logger.debug` lines inline in the code and let the user decide on the logging level?

Reply:
Mostly because the logging level is global.  I like to be able to target debug logging a bit, rather than just turning it on everywhere.

For example, if you had a system with 1000 watchers but only one that was having issues it'd probably be easier to debug if you could turn on debug logging just for that one.  Likewise, if you have a large system with many sub components, you might temporarily want logging for watchers without the pain of globally turning on debug logging.  (Though I suppose in this case you'd _have_ to set the global level to debug or you wouldn't get anything from the watchers either.  I guess this might need a bit more thought).

Reply:
Good point.  Would it be more flexible to be able to pass through the log level in the same way a Phoenix controller works? 
https://hexdocs.pm/phoenix/Phoenix.Controller.html#module-options

Reply:
Yeah, feels like that might be a better approach.